calico.yaml

```diff
@@ -0,0 +1,442 @@
+package:
+  name: calico
+  version: 3.26.0
+  epoch: 1
+  description: "Cloud native networking and network security"
+  target-architecture:
+    - x86_64
+    - aarch64
+  copyright:
+    - license: Apache-2.0
+
+environment:
+  contents:
+    packages:
+      - busybox
+      - ca-certificates-bundle
+      - build-base
+      - bash
+      - go
+      # for felix
+      - elfutils-dev
+      - zlib-dev
+      - iproute2
+      - zstd-dev
+      - llvm15
+      - clang-15
+      - libbpf
+      - linux-headers
+      # for BIRD
+      - flex
+      - bison
+      - autoconf
+      - automake
+      # TODO: Can remove this when cni-plugins are built from source
+      - curl
+
+# The calico build process is heavily tailored towards using Makefiles to
+# orchestrate docker commands. This makes for a less than ideal
+# experience replicating it in melange. To try to maintain some level of
+# sanity while packaging and updating calico, the approach here chooses to
+# use the upstream Makefile whenever possible, and only when necessary,
+# use the equivalent evaluated `docker run ...` commands within
+# independent pipeline blocks. Throughout this file you'll see the relevant
+# comments denoting the equivalent Makefile targets that each pipeline
+# block is trying to match, if applicable.
+pipeline:
+  - uses: git-checkout
+    with:
+      repository: https://github.com/projectcalico/calico
+      tag: v${{package.version}}
+      expected-commit: 8b103f46fbdc989e59d81e08d215ab4a59fa6cec
+  - runs: |
+      # Remediate GHSA-hqxw-f8mx-cpmw
+      go get github.com/docker/distribution@v2.8.2-beta.1
+      go mod tidy
+  - working-directory: felix
+    pipeline:
+      # Equivalent to target: "build-bpf"
+      - runs: |
+          # TODO: this is a terrible hack to prevent the error:
+          #   /usr/include/gnu/stubs.h:7:11: fatal error: 'gnu/stubs-32.h' file not found
+          # I have no idea why it's wanting stubs-32.h and not correctly recognizing stubs-64.h instead
+          sed -i 's/#include <gnu\/stubs\.h>/\n/' /usr/include/features.h
+
+          # TODO: Alternatively, we could patch the unused variable out...
+          export CFLAGS="-O2 -Wall -fomit-frame-pointer -march=x86-64-v2 -Wp,-D_FORTIFY_SOURCE=3"
```

Comment:
```suggestion
          export CFLAGS="$CFLAGS -Wno-error=unused-but-set-variable"
```

Though since this is BPF bytecode we probably don't really want the host CFLAGS anyway.

Reply:
You might also try `-ffreestanding` option for CFLAGS here to avoid the stubs-32.h hack.

Reply:
I followed the stack trace down to the `__x86_64__` macro not being set properly during our build, causing `/usr/include/gnu/stubs.h` to choose the `gnu/stubs-32.h` leg instead of the appropriate `x86_64` leg. to get around this I instead manually set `-D__x86_64__` as a compiler arg. this feels less hacky to me than removing the import `gnu/stubs` import entirely, but correct me if I'm wrong!